File: 0110-balanced-binary-tree/0110-balanced-binary-tree.php
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     public $val = null;
 *     public $left = null;
 *     public $right = null;
 *     function __construct($val = 0, $left = null, $right = null) {
 *         $this->val = $val;
 *         $this->left = $left;
 *         $this->right = $right;
 *     }
 * }
 */
class Solution {

    /**
     * @param TreeNode $root
     * @return Boolean
     */
    function isBalanced($root) {    
        $isBalanced = true;
        $this->getTreeDepth($root, $isBalanced);
        
        return $isBalanced;
    }
    
    public function getTreeDepth($node, &$isBalanced){
        if($node === null){
            return;
        }
        
        $leftDepth = $this->getTreeDepth($node->left, $isBalanced);
        $rightDepth = $this->getTreeDepth($node->right, $isBalanced);
        
        if(abs($leftDepth - $rightDepth) > 1){
            $isBalanced = false;
            return;
        }
        
        return max($leftDepth, $rightDepth) + 1;
    }
}

File: 0110-balanced-binary-tree/0110-balanced-binary-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func isBalanced(root *TreeNode) bool {
    if root == nil {
        return true
    }
    
    return abs(depth(root.Left) - depth(root.Right)) <= 1 && isBalanced(root.Left) && isBalanced(root.Right)
}

func depth(node *TreeNode) int {
    if node == nil {
        return 0
    }
    
    return max(depth(node.Left), depth(node.Right)) + 1
}

func abs(a int) int {
    if a < 0 {
        return -a
    }
    
    return a
}

func max(a, b int) int {
    if a > b {
        return a
    }
    
    return b
}